### PR TITLE
Remove 2.1 nightly images

### PR DESCRIPTION
**Pull request by natemcmaster**
HTTP 301 Moved Permanently
Location: https://github.com/dotnet/dotnet-docker/pull/457

As a part of #403, we can stop producing the microsoft/aspnetcore images for the 2.1 runtime. These images will be replaced by `microsoft/dotnet:2.1-aspnetcore-runtime`. 

This also deprecates `microsoft/aspnetcore-build:2.1` in favor of `microsoft/dotnet:2.1-sdk`. 

See https://github.com/aspnet/Announcements/issues/298 for details.

### CODE REVIEW

**Comment by natemcmaster** [comment]
@dotnet-bot test ci please

File: build-pipeline/pipelinejobs.groovy [hunk 1]
```diff
@@ -3,24 +3,21 @@ import org.dotnet.ci.pipelines.Pipeline
...
 }
 
 // Images that can only build on RS3, not RS1
-['2.0/nanoserver-1709/*', '2.1/nanoserver-1709/*'].each { folderFilter ->
-    def triggerName = "Windows 1709 ${folderFilter[0..-3]} Build"
-
-    windowsPipeline.triggerPipelineOnEveryGithubPR(triggerName, ['RS3':true, 'folderFilter':folderFilter])
-    windowsPipeline.triggerPipelineOnPush(triggerName, ['RS3':true, 'folderFilter':folderFilter])
-}
+def triggerName = "Windows 1709 2.0/nanoserver-1709 Build"
```
Comment (MichaelSimons) [inline build-pipeline/pipelinejobs.groovy]: I would guess moving `def triggerName` here is going to cause an issue with the next `triggerName` declaration

Comment (natemcmaster) [reply build-pipeline/pipelinejobs.groovy]: thanks!

**Comment by natemcmaster** [comment]
@dotnet-bot test ci please

**Pull request merged**